src/node/hooks/express/padurlsanitize.js

```diff
@@ -11,7 +11,13 @@ exports.expressCreateServer = (hookName, args, cb) => {
       return;
     }
 
-    const sanitizedPadId = await padManager.sanitizePadId(padId);
+    let sanitizedPadId;
+    try {
+      sanitizedPadId = await padManager.sanitizePadId(padId);
+    } catch {
+      res.status(500).send('Query inactivity timeout');
+      return;
+    }
```

Comment:
Let Express handle the error by passing it to `next`:

```javascript
  args.app.param('pad', (req, res, next, padId) => {
    (async () => {
      // ensure the padname is valid and the url doesn't end with a /
      if (!padManager.isValidPadId(padId) || /\/$/.test(req.url)) {
        res.status(404).send('Such a padname is forbidden');
        return;
      }

      const sanitizedPadId = await padManager.sanitizePadId(padId);

      if (sanitizedPadId === padId) {
        // the pad id was fine, so just render it
        next();
      } else {
        // the pad id was sanitized, so we redirect to the sanitized version
        const realURL =
            encodeURIComponent(sanitizedPadId) + new URL(req.url, 'http://invalid.invalid').search;
        res.header('Location', realURL);
        res.status(302).send(`You should be redirected to <a href="${realURL}">${realURL}</a>`);
      }
    })().catch((err) => next(err || new Error(err)));
  });
```

Reply:
Actually I tried this first and it didn't work, but yes your proposed change is definitely the idiomatic one. Will do.